lib/core/src/Cardano/Wallet/Api.hs

```diff
@@ -146,3 +158,33 @@ instance Accept Any where
         [ "application" // "json"
         , "application" // "json" /: ("charset", "utf-8")
         ]
+
+-- | A range of dates in ISO-8601 UTC format without symbols. Meant to be
+-- rendered as a HTTP 'Header', where the 'name' type-parameter renders as a
+-- prefix, e.g.
+--
+-- name 20190227T160329Z-*
+--
+-- 'Nothing' ("*") can be used instead of upper and/or lower boundary.
+--
+-- - `20190227T160329Z-*`: means all transactions after
+--    2019-02-27 T 16:03:29Z (including)
+-- - `*-20190227T160329Z`: means all transactions before 2019-02-27 T 16:03:29Z
+--    (including)
+-- - `*-*`: means all transactions
+-- - `20190227T000000Z-20200227T000000Z`: means all transaction between
+--    2019-02-27 and 2020-02-27, in ascending order.
+-- - `20200227T000000Z-20190227T000000Z`: means all transaction between
+--  2020-02-27 and 2019-02-27, in descending order.
+data Iso8601Range (name :: Symbol)
+    = Iso8601Range (Maybe UTCTime) (Maybe UTCTime)
+    deriving (Show)
+
+instance FromHttpApiData (Iso8601Range (name :: Symbol)) where
+    parseHeader = error "FromHttpApiData Iso8601Range to be implemented"
+    parseQueryParam = error "I guess we need to implement this too?"
```

Comment:
No we don't :)
Only one of the methods needs to be implemented, they're all implemented using one or the other. Note that this is the semantic behind "minimal complete definition" in the hackage documention:

https://hackage.haskell.org/package/http-api-data-0.4.1/docs/Web-HttpApiData.html#t:FromHttpApiData

Here it says:

<pre>
<b>Minimal complete definition</b>

<code>parseUrlPiece | parseQueryParam</code>
</pre>

Which means that you need to implement either `parseUrlPiece` or `parseQueryParam` and, other methods of the class are implicitly implemented :)

Reply:
@KtorZ I looked at that! But here we want to implement **`parseHeader`**, which is **not** `parseUrlPiece` nor `parseQueryParam` and thus not a minimal complete implementation. And if you try anyway, it errors as expected. 😕

Reply:
Na, you can get away by just implementing `parseUrlPiece`. `parseHeader` in the end is just:

```
class FromHttpApiData a where
  {-# MINIMAL parseUrlPiece | parseQueryParam #-}
  -- | Parse URL path piece.
  parseUrlPiece :: Text -> Either Text a
  parseUrlPiece = parseQueryParam

  -- | Parse HTTP header value.
  parseHeader :: ByteString -> Either Text a
  parseHeader = parseUrlPiece <=< (left (T.pack . show) . decodeUtf8')

  -- | Parse query param value.
  parseQueryParam :: Text -> Either Text a
  parseQueryParam = parseUrlPiece
```

It's really just an alias. As long as you implement `parseUrlPiece`, you're fine.

Reply:
Ah. Thanks.